react/src/utils/Path.js

```diff
@@ -0,0 +1,28 @@
+// -------------------------------------------------------------------------------------------------
+// Path related function.
+// -------------------------------------------------------------------------------------------------
+
+import Str from './Str';
+
+function GetNormalizedPath(path) {
+    if (!Str.HasValue(path)) {
+        return "";
+    }
+    path = path.replace(/\/+/g, "/");
+    if (path.endsWith("/")) {
+        path = path.substring(0, path.length - 1)
+    }
+    if (!path.startsWith("/")) {
+        path = "/" + path;
```

Comment:
Is this turning a relative path into an absolute one? is it only used in certain contexts? How would I know?

Reply:
No not absolute path, all paths on client side are relative, so GetPath("/users", "some-env") == "/api/react/some-env/users" ... So the call doesn't have to know that "/api/react" is the prefix of the React UI paths.